collector/test/CollectorConfigTest.cpp

```diff
@@ -32,6 +34,10 @@ class MockCollectorConfig : public CollectorConfig {
   void MockSetEnableExternalIPs(bool value) {
     SetEnableExternalIPs(value);
   }
+
+  bool MockYamlConfigToConfig(YAML::Node& yamlConfig) {
+    return YamlConfigToConfig(yamlConfig);
+  }
```

Comment:
Same question I asked previously, why do we need a mock method if we are just calling the real method? You should be able to instantiate a `CollectorConfig` in the tests, instead of a `MockCollectorConfig`.
```suggestion
```

Reply:
`YamlConfigToConfig` is protected, so I do need to use the mock.

Reply:
No, you don't, you can just use a `FRIEND_TEST`
http://google.github.io/googletest/reference/testing.html#FRIEND_TEST

You can find an example on how to use it here:
https://github.com/stackrox/collector/blob/0cbc2f0a2846122d203495e66815be8c0dc20832/collector/lib/system-inspector/Service.h#L47
https://github.com/stackrox/collector/blob/0cbc2f0a2846122d203495e66815be8c0dc20832/collector/test/SystemInspectorServiceTest.cpp#L8

Reply:
I see the whole test file is doing this though, so I'll just resolve the comment and refactor the thing in a follow up.